### DISCUSSION TITLE
`m-[1rem 2rem]` instead of `m-[1rem_2rem]`

### DISCUSSION THREAD

**Discussion by babakfp** [tailwindlabs/tailwindcss#9443] [Ideas]
Hi
I would like to be able to do this:

```HTML
<div class="m-[1rem 2rem]">Hello, World!</div>

<style>
  [class*="m-[1rem 2rem]"] {
    margin: 1rem 2rem;
  }
</style>
```

This is how it works currently (the underline `_`):

```HTML
<div class="m-[1rem_2rem]">Hello, World!</div>
```

Example in Playground:
https://play.tailwindcss.com/kLXrvJqXYb?file=css

## Problems with this idea

Something like this could be problematic: https://play.tailwindcss.com/YxX0xl5eGE?file=css

```HTML
<div class="bg-[ black ]">Hello, World!</div>

<style>
  [class*="bg-[ black ]"] {
    background: gray;
  }
  .black {
    background: blue !important;
  }
</style>
```

Also, I don't know about the problem compatibility stuff related to this solution.

That's all. Thanks.

**Comment by olets** [comment]
Classes are space-delimited: in `class="m-[1rem 2rem]"` there are two classes, `m-[1rem` and `2rem]`. (The second is invalid, because classes can't start with a number.)

You're allowed to use `[` and `]` in classes, so Tailwind would have no way of knowing whether `[`/`]` was related to an arbitrary Tailwind class or just part of a class.

Your example of what you want to do will work if you don't use Tailwind. https://jsfiddle.net/mwdeh7nL/ (Though the CSS selector would typically be written `.bg-\[.black.\]`)

In your problem example, you have three classes - `bg-[`, `black`, and `]`. Why not `<div class="bg-[gray]">Hello, World!</div>`?
